lib/ProductOpener/Display.pm

```diff
@@ -7480,124 +7455,42 @@ HTML
 				$html_upc .= " " . $' . " (UPC / UPC-A)";
 			}
 		}
-		$html .= "<p id=\"barcode_paragraph\">"
-		    . lang("barcode") . separator_before_colon($lc)
-		    . ": <span id=\"barcode\" property=\"food:code\" itemprop=\"gtin13\" style=\"speak-as:digits;\">$code</span> $html_upc</p>
-<div property=\"gr:hasEAN_UCC-13\" content=\"$code\" datatype=\"xsd:string\"></div>\n";
+		$template_data_ref->{html_upc} = $html_upc;
 	}
 
-
 	# obsolete product
 
 	if ((defined $product_ref->{obsolete}) and ($product_ref->{obsolete})) {
-
+		$template_data_ref->{product_ref_obsolete} = $product_ref->{obsolete};
```

Comment:
`product_is_obsolete` might be a better key name. `$product_ref` is just named like this since it's a reference variable.